### DISCUSSION TITLE
Separate out Workflow controller and executor

### DISCUSSION THREAD

**Discussion by liafizan** [argoproj/argo-workflows#6980] [General]
Is it possible to have workflow controller and executor run in separate clusters but communicate via a message bus instead? We want to dedicate a workflow controller for customer but then have workflows run in a different cluster that customer owns? ( Use of message bus is because there is no K8S api access between those clusters )

**Comment by sarabala1979** [comment]
We proposed https://github.com/argoproj/argo-workflows/issues/3523 for multi cluster/multi namespace support.  But it depends on k8s API.
